### PR TITLE
create-repo-hw-solution

### PR DESCRIPTION
**Pull request by margosha99**
Configuration files added, Spring Boot project with Book entity created

### CODE REVIEW

File: src/main/java/com/example/demo/service/BookServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+public class BookServiceImpl implements BookService {
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Comment (okuzan) [inline src/main/java/com/example/demo/service/BookServiceImpl.java]: add implementations

File: src/main/java/com/example/demo/service/BookService.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.demo.service;
+
+import com.example.demo.model.Book;
+import java.util.List;
+
+public interface BookService {
+    Book save(Book book);
+
+    List<Book> findAll();
+
```
Comment (okuzan) [inline src/main/java/com/example/demo/service/BookService.java]:
```suggestion
```

File: src/main/java/com/example/demo/repository/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+public class BookRepositoryImpl implements BookRepository {
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Comment (okuzan) [inline src/main/java/com/example/demo/repository/BookRepositoryImpl.java]: add impl

File: src/main/java/com/example/demo/repository/BookRepository.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.example.demo.model.Book;
+import java.util.List;
+
+public interface BookRepository {
+
+    Book save(Book book);
+
+    List<Book> findAll();
+
```
Comment (okuzan) [inline src/main/java/com/example/demo/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/com/example/demo/repository/BookRepository.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.example.demo.repository;
+
+import com.example.demo.model.Book;
+import java.util.List;
+
+public interface BookRepository {
+
```
Comment (okuzan) [inline src/main/java/com/example/demo/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/com/example/demo/model/Book.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (okuzan) [inline src/main/java/com/example/demo/model/Book.java:21]:
```suggestion
```

File: src/main/java/com/example/demo/model/Book.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (okuzan) [inline src/main/java/com/example/demo/model/Book.java]: Don't use @Data on JPA entities, they contain @ToString and @EqualAndHashcode that can be problematic, use @ Getter @Setter here

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**